src/hotspot/cpu/aarch64/stubGenerator_aarch64.cpp

```diff
@@ -5488,6 +5474,23 @@ class StubGenerator: public StubCodeGenerator {
 
     __ reset_last_Java_frame(true);
 
+    BarrierSetAssembler* bs_asm = BarrierSet::barrier_set()->barrier_set_assembler();
+
+    // In case a concurrent thread disarmed the nmethod, we need to ensure the new instructions
```

Comment:
```suggestion
    // In case a concurrent thread disarmed this nmethod, we need to ensure the new instructions
```
I think you mean this nmethod?